pkg/envoy/registry/services.go

```diff
@@ -50,6 +50,8 @@ func (k *KubeProxyServiceMapper) ListProxyServices(p *envoy.Proxy) ([]service.Me
 	}
 
 	meshServices := kubernetesServicesToMeshServices(k.KubeController, services)
+	// filter down meshServices to eliminate subdomains not from this pod (keeping empty subdomains)
+	meshServices = service.FilterMeshServicesBySubdomain(meshServices, pod.GetName(), true)
```

Comment:
Instead of this, I think we can get 2 birds one stone?

Both `kubernetesServicesToMeshServices`, and `listServicesForPod` only exist in one place. Can we instead have `listServicesForPod` return meshServices, appropriately filtered already?

Reply:
The problem is that a single headless service can spawn N MeshServices where N = # of endpoints in that headless service. So from the k8s service perspective, there's no way to pre-filter

Reply:
Ah, I'm with you now; that makes sense 👍🏾